examples/options/src/track_propagation.cpp

```diff
@@ -0,0 +1,72 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "traccc/options/track_propagation.hpp"
+
+// Detray include(s).
+#include <detray/definitions/units.hpp>
+
+// System include(s).
+#include <limits>
+
+namespace traccc::opts {
+
+/// Convenience namespace shorthand
+namespace po = boost::program_options;
+
+track_propagation::track_propagation()
+    : interface("Track Propagation Options") {
+
+    m_desc.add_options()("constraint-step-size-mm",
+                         po::value(&(config.stepping.step_constraint))
+                             ->default_value(std::numeric_limits<float>::max()),
+                         "The constrained step size [mm]");
+    m_desc.add_options()("overstep-tolerance-um",
+                         po::value(&(config.navigation.overstep_tolerance))
+                             ->default_value(-100.f),
+                         "The overstep tolerance [um]");
+    m_desc.add_options()(
+        "mask-tolerance-um",
+        po::value(&(config.navigation.mask_tolerance))->default_value(15.f),
+        "The mask tolerance [um]");
+    m_desc.add_options()(
+        "search-window",
+        po::value(&m_search_window)->default_value(m_search_window),
```

Comment:
This relates back to what I wrote about having to use a custom type for custom "read rules".

:thinking: But your question did give me an idea. As I also didn't quite like how I went about this variable. Let me try to create a sibling to `traccc::opts::value_array` that could streamline this thing. Since I'd want to have similar variables in other places as well. (Where we configure "config types" directly from the command line with minimal overhead in coding needed.)

Reply:
Oh, I see. Didn't know that.

Reply:
I tried being fancy here for a bit. :thinking: I thought that I could come up with a type that could "wrap" an existing `std::array` variable, to expose it to the command line parsing like that. But unfortunately `boost::program_options` does everything by value semantics. So the following didn't work:

```c++
/// Wrapper around an array, to allow for command line settings
///
/// To be able to expose an "existing array" (from let's say a configuration
/// type) to the command line parsing, this type can be used as a wrapper. A
/// helper object still needs to be created, but at least configured values get
/// set directly on the array in question.
///
template <class ARRAY_TYPE>
struct value_array_wrapper {

    /// The wrapped array
    ///
    /// @note It's a pointer and not let's say an @c std::reference_wrapper,
    ///       because the object needs to be default constructible. (Because
    ///       of how Boost.Program_options works.)
    ///
    ARRAY_TYPE* array = nullptr;

};  // class value_array_wrapper
```

As the Boost code would create such objects itself. Without the pointer being set correctly on them. :frowning:

So, we need to stay with the current setup for now...